Frameworks/UIKit.Xaml/TextField.xaml.cpp

```diff
@@ -0,0 +1,759 @@
+Canvas^ TextField::SublayerCanvas::get() {
+    return nullptr;
+}
+
+// Accessor for the LayerProperty that manages the BorderBrush of this TextField
+Private::CoreAnimation::LayerProperty^ TextField::GetBorderBrushProperty() {
+    return nullptr;
+}
+
+// Accessor for the LayerProperty that manages the BorderThickness of this TextField
+Private::CoreAnimation::LayerProperty^ TextField::GetBorderThicknessProperty() {
+    return nullptr;
+}
+
+TextBox^ TextField::TextBox::get() {
+    return safe_cast<Windows::UI::Xaml::Controls::TextBox^>(_backingControl);
+}
+
+PasswordBox^ TextField::PasswordBox::get() {
+    return safe_cast<Windows::UI::Xaml::Controls::PasswordBox^>(_backingControl);
+}
+
+void TextField::KillFocus() {
+    if (SecureEntry) {
+        PasswordBox->IsEnabled = false;
+        PasswordBox->IsEnabled = true;
+    } else {
+        TextBox->IsEnabled = false;
+        TextBox->IsEnabled = true;
+    }
+}
+
+void TextField::SetFocus() {
+    if (SecureEntry) {
+        PasswordBox->Focus(FocusState::Programmatic);
+    } else {
+        TextBox->Focus(FocusState::Programmatic);
+    }
+}
+
+bool TextField::BecomeFirstResponder() {
+    if (SecureEntry) {
+        return PasswordBox->Focus(FocusState::Programmatic);
+    } else {
+        return TextBox->Focus(FocusState::Programmatic);
+    }
+}
+
+void TextField::OnTextChanged(Platform::Object ^sender, Windows::UI::Xaml::Controls::TextChangedEventArgs ^e)
+{
+    if (_textChangedHandler) {
+        _textChangedHandler->Invoke(
+            InspectableFromObject(this).Get(),
+            ObjectToType<ABI::Windows::UI::Xaml::IRoutedEventArgs>(e).Get());
+    }
+}
+
+void TextField::OnPasswordChanged(Platform::Object ^sender, Windows::UI::Xaml::RoutedEventArgs ^e)
+{
+    if (_textChangedHandler) {
+        _textChangedHandler->Invoke(
+            InspectableFromObject(this).Get(),
+            ObjectToType<ABI::Windows::UI::Xaml::IRoutedEventArgs>(e).Get());
+    }
+}
+
+void TextField::OnGotFocus(Platform::Object ^sender, Windows::UI::Xaml::RoutedEventArgs ^e)
+{
+    if (_gotFocusHandler) {
+        _gotFocusHandler->Invoke(
+            InspectableFromObject(this).Get(),
+            ObjectToType<ABI::Windows::UI::Xaml::IRoutedEventArgs>(e).Get());
+    }
+}
+
+void TextField::OnLostFocus(Platform::Object ^sender, Windows::UI::Xaml::RoutedEventArgs ^e)
+{
+    if (_lostFocusHanlder) {
+        _lostFocusHanlder->Invoke(
+            InspectableFromObject(this).Get(),
+            ObjectToType<ABI::Windows::UI::Xaml::IRoutedEventArgs>(e).Get());
+    }
+}
+
+void TextField::OnKeyDown(Platform::Object ^sender, Windows::UI::Xaml::Input::KeyRoutedEventArgs ^e)
+{
+    if (_enterKeyDownHandler) {
+        if (e->Key == Windows::System::VirtualKey::Enter) {
+            _enterKeyDownHandler->Invoke(
+                InspectableFromObject(this).Get(),
+                ObjectToType<ABI::Windows::UI::Xaml::IRoutedEventArgs>(e).Get());
+            e->Handled = true;
```

Comment:
we don't want to always handle it, do we?  What if the user actually wants to type enter and the delegate doesn't kill focus when enter is clicked?